### PR TITLE
Async loading

### PR DESCRIPTION
**Pull request by ivan-kleshnin**
Excellent! Pls. see my comments below for a minor issues.

### CODE REVIEW

File: themes/paqmind/components/codemirror.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+let $ = require("jquery");
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/codemirror.js]: `require` vs `import`?

File: themes/paqmind/components/codemirror.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  "clojure": ["clj", "clojure"],
+  "javascript": ["js", "javascript", "es"],
+  "jinja2": ["jinja2"],
+  "haskell": ["hs", "haskell"],
+  "htmlmixed": ["html"],
+  "python": ["py", "python"],
+  "xml": ["xml"],
+};
+
+let findMode = function (classes) {
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/codemirror.js]: Cool. But why not just

```js
let findMode = function (array) {
  return find(key => intersection(array, codemirrorModes[key]).length, keys(codemirrorModes));
};
```

If we define `codemirrorModes` why check for empty key?

File: themes/paqmind/components/codemirror.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  return find(
+    function(key) {
+      return key ? intersection(classes, codemirrorModes[key]).length : false;
+    }, keys(codemirrorModes));
+};
+
+let getClassList = (elem) => $(elem).attr("class") ? $(elem).attr("class").split(/\s+/) : [];
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/codemirror.js]: Too complex line. Use a common `function` ?

File: themes/paqmind/components/codemirror.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+let $codeElems = $("code");
+if ($codeElems.length) {
+  let modes = uniq(reduce((memo, elem) => {
+    let result = findMode(getClassList(elem));
+    if (result) memo.push(result);
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/codemirror.js]: Use `append` instead of `.push`.

File: themes/paqmind/components/codemirror.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    return memo;
+  }, [], $codeElems));
+
+  if (modes.length) {
+    require.ensure([], function(require) {
+      require("codemirror/lib/codemirror.css");
+      require("../less/site/components/cm-theme-paqmind.less");
+      require("codemirror/lib/codemirror");
+      let CodeMirror = require("codemirror");
+      modes.map((mode) => {require(`codemirror/mode/${mode}/${mode}`);});
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/codemirror.js]: `map` implies some result. Here we just produce side-effects so it's better to use `each` or `forEach`.

File: themes/paqmind/components/modal.js [hunk 6]
```diff
@@ -1,44 +1,35 @@
...
+    let animation = $modal.data('animation');
 
-    toggle.addEventListener('click', () => {
-      event.preventDefault();
-      modal.classList.remove("hide");
-      body.className += " noscroll";
-      if (animated) {
-        if (animationHide) modal.classList.remove(animationHide);
-        if (animationShow) modal.classList.add(animationShow);
+    let toggleModal = () => {
```
Comment (ivan-kleshnin) [inline themes/paqmind/components/modal.js]: I think it's better to use common `function` here. More readable.

**Comment by scabbiaza** [comment]
Thank you for the review! I learned a lot.
Pushed changes.

**Pull request merged**